pkg/controller/scyllacluster/resource.go

```diff
@@ -1125,3 +1190,270 @@ func MakeJobs(sc *scyllav1.ScyllaCluster, services map[string]*corev1.Service, i
 
 	return jobs, progressingConditions
 }
+
+func MakeEndpointSlicesAndEndpoints(sc *scyllav1.ScyllaCluster, services map[string]*corev1.Service, podLister corev1listers.PodLister) ([]*discoveryv1.EndpointSlice, []*corev1.Endpoints, error) {
+	var endpointSlices []*discoveryv1.EndpointSlice
+	var endpoints []*corev1.Endpoints
+
+	endpointPortsHash := func(name string, ports []discoveryv1.EndpointPort) (string, error) {
```

Comment:
correct me if I'm wrong but altogether this is hashing with sha512, encoding with base64 and then converting to hex
```shell
$ echo -n "test" | sha512sum - |  cut -f 1 -d " " | base64 -w0 - | od -A n -t x1 | sed 's/ *//g' | tr -d '\n' | wc -m
344
```
by taking just the first 8 characters of the result string, aren't you significantly reducing the result space and running not that improbable of a risk of collision?